group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1511,7 +1538,12 @@ public void replay(
         String groupId = key.groupId();
         String memberId = key.memberId();
 
-        ConsumerGroup consumerGroup = getOrMaybeCreateConsumerGroup(groupId, value != null);
+        ConsumerGroup consumerGroup;
+        if (value != null) {
+            consumerGroup = maybeCreateAndUpdateConsumerGroup(groupId);
+        } else {
+            consumerGroup = getOrMaybeCreateConsumerGroup(groupId, false);
```

Comment:
It is a bit weird to keep using the old method in replay methods. If we do my previous suggestion, we may be able to use the new one everywhere. What do you think?